cmd/lakectl/cmd/doctor.go

```diff
@@ -11,11 +11,42 @@ import (
 	"github.com/treeverse/lakefs/pkg/api"
 )
 
-var (
-	ErrCredential       = errors.New("credential error")
-	ErrWrongEndpointURI = errors.New("wrong endpoint_uri error")
-	ErrUnknownConfig    = errors.New("unknown configuratioin error")
-)
+type ErrCredential struct {
+	Message string
+	Details string
+}
+
+func (e *ErrCredential) Error() string {
+	return (e.Message + "\n" + e.Details + "\n")
```

Comment:
Error strings should not end in a newline.  In this case a more Go-ish version would say e.g.

```go
type Detailed interface {
	Details() string
}


type CredentialsError struct {
	Message string
	Details string
}

func (e *CredentialsError) Error() string { return e.Message }

func (e *CredentialsError) Details() string {
	return e.Message + "\n" + e.Details
}
```

and then when printing an error you could see whether it has details by saying e.g.

```go
if detailedErr, ok := err.(Detailed); ok {
	Write(DetailedErrorTemplate, detailedErr)
} else {
	Write(ErrorTemplate, err)
}
```

Now `DetailedErrorTemplate` can use the `Details` field (even though it never calls the function, go figure...).